core/tests/puppeteer-acceptance-tests/utilities/user/logged-in-user.ts

```diff
@@ -157,6 +200,162 @@ export class LoggedInUser extends BaseUser {
       throw newError;
     }
   }
+
+  /**
+   * Clicks the delete account button and waits for navigation.
+   */
+  async deleteAccount(): Promise<void> {
+    await this.clickAndWaitForNavigation(deleteAccountButton);
+  }
+
+  /**
+   * Clicks on the delete button in the page /delete-account to confirm account deletion, also, for confirmation username needs to be entered.
+   * @param {string} username - The username of the account.
+   */
+  async confirmAccountDeletion(username: string): Promise<void> {
+    await this.clickOn(accountDeletionButtonInDeleteAccountPage);
+    await this.type(confirmUsernameField, username);
+    await this.clickAndWaitForNavigation(confirmAccountDeletionButton);
+  }
+
+  /**
+   * Navigates to the sign up page. If the user hasn't accepted cookies, it clicks 'OK' to accept them.
+   * Then, it clicks on the 'Sign in' button.
+   */
+  async navigateToSignUpPage(): Promise<void> {
+    await this.goto(testConstants.URLs.Home);
+    if (!this.userHasAcceptedCookies) {
+      await this.clickOn('OK');
+      this.userHasAcceptedCookies = true;
+    }
+    await this.clickOn('Sign in');
+  }
+
+  /**
+   * Clicks on the link to the Oppia Wiki, which opens in a new tab.
+   */
+  async clickAdminAccessInfoLink(): Promise<void> {
+    await this.clickLinkAnchorToNewTab(
+      'Oppia Wiki',
+      WikiPrivilegesToFirebaseAccount
+    );
+  }
+
+  /**
+   * Enters the provided username into the sign up username field and sign in if the username is correct.
+   * @param {string} username - The username to enter.
+   */
+  async signInWithUsername(username: string): Promise<void> {
+    await this.clearAllTextFrom(signUpUsernameField);
+    await this.type(signUpUsernameField, username);
+    // Using blur() to remove focus from signUpUsernameField.
+    await this.page.evaluate(selector => {
+      document.querySelector(selector).blur();
+    }, signUpUsernameField);
+
+    await this.waitForPageToFullyLoad();
+    const invalidUsernameErrorContainerElement = await this.page.$(
+      invalidUsernameErrorContainer
+    );
+    if (!invalidUsernameErrorContainerElement) {
+      await this.clickOn('input.e2e-test-agree-to-terms-checkbox');
```

Comment:
Set as constants, also for the rest below.

Reply:
done.